src/core/PAGE_main.jl

```diff
@@ -339,7 +339,8 @@ function perturb_marginal_page_emissions!(base::Model, marginal::Model, gas::Sym
         update_param!(marginal.mi.md, :add, forcing_shock)
     end
 
-    return nothing
+    # return a new model from the marginal model's model instance
+    return Model(marginal.mi)
```

Comment:
@ckingdon95 This is the minimal code to fix the erroring SCC calculation, however it is admittedly perhaps not the cleanest way.  Basically the way we treat `m.md` and `m.mi.md` in `perturb_marginal_page_emissions!` isn't working with the current flow of data, such that the call 
```
if year != nothing
        run(base)
        Mimi.build!(marginal)
        marginal = perturb_marginal_page_emissions!(base, marginal, gas, year)
        run(marginal)
    end
``` 
in `get_marginal_page_models` has no side effects, run(marginal)` before or after this block makes no change to the `marginal[:EquityWeighting, :td_totaldiscountedimpacts]` value